### PR TITLE
[subinterface]Added additional checks in portchannel and subinterface commands

### PR DESCRIPTION
**Pull request by dgsudharsan**
#### What I did
Added additional checks in subinterface and portchannel commands so they don't conflict. Without the checks, a subinterface could be created on a portchannel member and vice versa which will lead to SAI failure followed by orchagent crash.

#### How I did it
Added checks in relevant config functions.

#### How to verify it
Added UT to verify it.

### CODE REVIEW

**Comment by dgsudharsan** [comment]
@prsunny @preetham-singh  Can you please review?

**Comment by liat-grozovik** [comment]
@yxieca  we need this fix for 202205, please help to prioritize the review

**Comment by yxieca** [comment]
@wendani can you also help reviewing?

File: tests/subintf_test.py [hunk 1]
```diff
@@ -141,6 +147,21 @@ def test_invalid_subintf_creation(self):
         print(result.exit_code, result.output)
         assert result.exit_code != 0
 
+    def test_subintf_creation_on_lag_member(self):
```
Comment (preetham-singh) [inline tests/subintf_test.py:150]: Inline is it possible to add test to check creation of subintf on vlan members?

Comment (dgsudharsan) [reply tests/subintf_test.py:150]: That's a different functionality check not related to this issue. I recommend you to raise it if it is missed.

Comment (preetham-singh) [reply tests/subintf_test.py:150]: Agreed. Lets track it separately for negative scenario of subintf creation on vlan member.

**Review by preetham-singh — APPROVED** [review]

**Review by prsunny — APPROVED** [review]

**Pull request merged**